int number_of_commands
int command